src/client/creature.cpp

```diff
@@ -510,19 +510,17 @@ void Creature::updateWalkAnimation(int totalPixelsWalked)
     if(m_outfit.getCategory() != ThingCategoryCreature)
         return;
 
-    int footAnimPhases = getAnimationPhases() - 1;
-    int footDelay = getStepDuration(true) / getAnimationPhases() + 15;
+    int footAnimPhases, footDelay;
     if (m_outfit.getMount() != 0) {
+        // Since mount is a different outfit we need to get the mount animation phases
         auto datType = g_things.rawGetThingType(m_outfit.getMount(), ThingCategoryCreature);
-        footAnimPhases = datType->getAnimationPhases() - 1;
-        if (datType->getIdleAnimator() != nullptr) {
-            footDelay = getStepDuration(true) / ceil(footAnimPhases / 2) + 15;
-        }
-    } else if (getIdleAnimator() != nullptr) {
-        footDelay = getStepDuration(true) / ceil(footAnimPhases / 2) + 15;
+        footAnimPhases = datType->getAnimationPhases();
+    } else {
+        footAnimPhases = getAnimationPhases();
     }
+    footAnimPhases = std::max<int>(0, footAnimPhases - 1);
+    footDelay = getStepDuration(true) / std::max<int>(1, footAnimPhases);
```

Comment:
Declare `footDelay` here instead:

```cpp
int footDelay = getStepDuration(true) / std::max<int>(1, footAnimPhases);`
```

Reply:
Sure, unfortunately this PR comes from repo which is no longer there. I tried but couldn't add news commits to this PR. Do you want me to open a new one with changes you requested?

Reply:
Yes, that would be good. This time it will not take as much to review :wink: